Bottom: 479ed0e031afd3ddeea4f9682750dc7532f1ac7a
Top:    6cdba82dd0ae52044f367af6a073124d65ca8614
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-05-03 16:49:38 +0100

Try a new, different sized VBO buffer for each upload.


---

diff --git a/src/hid/common/hidgl.c b/src/hid/common/hidgl.c
index f015fd1..990b017 100644
--- a/src/hid/common/hidgl.c
+++ b/src/hid/common/hidgl.c
@@ -135,10 +135,6 @@ hidgl_init_triangle_array (triangle_buffer *buffer)
    */
   buffer->use_map = false;
 
-  /* If using VBOs (but not mapping), we only need to this once */
-  if (buffer->use_vbo && !buffer->use_map)
-    glBufferData (GL_ARRAY_BUFFER, BUFFER_SIZE, NULL, GL_STREAM_DRAW);
-
   buffer->triangle_array = NULL;
   hidgl_reset_triangle_array (buffer);
 }
@@ -176,10 +172,10 @@ hidgl_flush_triangles (triangle_buffer *buffer)
       glUnmapBuffer (GL_ARRAY_BUFFER);
       buffer->triangle_array = NULL;
     } else {
-      /* NB: We only upload the portion of the buffer we've used */
-      glBufferSubData (GL_ARRAY_BUFFER, 0,
-                       BUFFER_STRIDE * buffer->vertex_count,
-                       buffer->triangle_array);
+      glBufferData (GL_ARRAY_BUFFER,
+                    BUFFER_STRIDE * buffer->vertex_count,
+                    buffer->triangle_array,
+                    GL_STREAM_DRAW);
     }
   } else {
     data_pointer = buffer->triangle_array;
